### ISSUE TITLE
Can't use Docker run with --chat-model

### ISSUE THREAD

**Issue by katopz** [TabbyML/tabby#685]
**Describe the bug**
Couldn't run with `--chat-model` args

## TL;DR
- ✅ `tabbyml/tabby` working `--model`
- ❌ `tabbyml/tabby` not working `--chat-model`
- ❌ `tabbyml/tabby:main-f15926f` not working `--model`
- ❌ `tabbyml/tabby:main-f15926f` not working `--chat-model`

### ❌ `tabbyml/tabby` not working `--chat-model`
```
sudo docker run -it --gpus all -p 8080:8080 -v $HOME/.tabby:/data tabbyml/tabby serve --model TabbyML/StarCoder-1B --chat-model TabbyML/Mistral-7B --device cuda
```

### Output and exit
```
2023-11-01T01:12:38.277398Z  INFO tabby::serve: crates/tabby/src/serve/mod.rs:143: Starting server, this might takes a few minutes...
2023-11-01T01:12:38.286822Z  INFO tabby::serve::search: crates/tabby/src/serve/search.rs:190: Index is ready, enabling server...
```

### ❌ `tabbyml/tabby:main-f15926f` not working `--model`
```
 sudo docker run -it --gpus all -p 8080:8080 -v $HOME/.tabby:/data tabbyml/tabby:main-f15926f serve --model TabbyML/StarCoder-1B --device cuda
```

### Output and exit
```
2023-11-01T01:52:04.164055Z  INFO tabby::serve: crates/tabby/src/serve/mod.rs:140: Starting server, this might takes a few minutes...
2023-11-01T01:52:04.166446Z  INFO tabby::serve::search: crates/tabby/src/serve/search.rs:190: Index is ready, enabling server...
ggml_init_cublas: GGML_CUDA_FORCE_MMQ:   no
ggml_init_cublas: CUDA_USE_TENSOR_CORES: yes
ggml_init_cublas: found 1 CUDA devices:
  Device 0: NVIDIA GeForce RTX 4090, compute capability 8.9
```

### ❌ `tabbyml/tabby:main-f15926f` not working `--chat-model`
```
sudo docker run -it --gpus all -p 8080:8080 -v $HOME/.tabby:/data tabbyml/tabby:main-f15926f serve --model TabbyML/StarCoder-1B --chat-model TabbyML/Mistral-7B --device cuda
```
### Output and exit
```
2023-11-01T01:47:03.151268Z  INFO tabby::serve: crates/tabby/src/serve/mod.rs:140: Starting server, this might takes a few minutes...
2023-11-01T01:47:03.157490Z  INFO tabby::serve::search: crates/tabby/src/serve/search.rs:190: Index is ready, enabling server...
ggml_init_cublas: GGML_CUDA_FORCE_MMQ:   no
ggml_init_cublas: CUDA_USE_TENSOR_CORES: yes
ggml_init_cublas: found 1 CUDA devices:
  Device 0: NVIDIA GeForce RTX 4090, compute capability 8.9
```

**Information about your version**
Please provide output of `tabby --version`
`0.4.0`

**Information about your GPU**
```
Wed Nov  1 08:15:59 2023
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.54.03              Driver Version: 535.98       CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 4090        On  | 00000000:01:00.0  On |                  Off |
|  0%   41C    P8              38W / 450W |   8582MiB / 24564MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+

+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A        12      G   /Xwayland                                 N/A      |
+---------------------------------------------------------------------------------------+
```

**Additional context**
I try to test chat api but can't get the server running with chat model.

**Comment by icycodes** [comment]
Hi, @katopz 
I noticed that the logs in the section `tabbyml/tabby not working --chat-model` do not contain any error messages. Is the process exited without printing any error logs?

**Comment by katopz** [comment]
> Hi, @katopz I noticed that the logs in the section `tabbyml/tabby not working --chat-model` do not contain any error messages. Is the process exited without printing any error logs?

Yes, just exit.

**Comment by katopz** [comment]
The exit code is `137` (and some `132`) so it cloud mean not enough memory.

But still no joy after set more memory.
```
katopz@shikuwa:~/.tabby$ sudo docker run --memory="16g" --memory-swap="16g" -it --gpus all -p 8080:8080 -v $HOME/.tabby:
/data tabbyml/tabby:main-f15926f serve --model TabbyML/StarCoder-1B --chat-model TabbyML/Mistral-7B --device cuda
2023-11-01T10:12:57.635943Z  INFO tabby::serve: crates/tabby/src/serve/mod.rs:140: Starting server, this might takes a few minutes...
ggml_init_cublas: GGML_CUDA_FORCE_MMQ:   no
ggml_init_cublas: CUDA_USE_TENSOR_CORES: yes
ggml_init_cublas: found 1 CUDA devices:
  Device 0: NVIDIA GeForce RTX 4090, compute capability 8.9
katopz@shikuwa:~/.tabby$ sudo docker ps -a
CONTAINER ID   IMAGE                        COMMAND                  CREATED          STATUS                        PORTS     NAMES
e49a293cb39f   tabbyml/tabby:main-f15926f   "/opt/tabby/bin/tabb…"   34 seconds ago   Exited (132) 21 seconds ago             nostalgic_galileo
315b90e1b4a8   tabbyml/tabby:latest         "/opt/tabby/bin/tabb…"   3 minutes ago    Exited (137) 3 minutes ago              tender_knuth
164c26529541   tabbyml/tabby                "/opt/tabby/bin/tabb…"   4 minutes ago    Exited (137) 3 minutes ago              jolly_chandrasekhar
857566087eeb   tabbyml/tabby                "/opt/tabby/bin/tabb…"   4 hours ago      Exited (137) 4 hours ago
      distracted_wilbur
katopz@shikuwa:~/.tabby$
```

**Comment by katopz** [comment]
### Update
1. It's working after i rebuild docker image to match my cuda version `12.0.0`.
2. It's working after updated to bdd9e6c
3. I'm not sure docker from docker hub is working or not (not test yet and i'm happy with local build now).

```
CUDA_VERSION=12.2.0 docker build -t tabby_cuda12_2 .
docker run -m 8g -it --gpus all -p 8080:8080 -v $HOME/.tabby:/data tabby_cuda12_2 serve --device cuda --model TabbyML/StarCoder-1B --chat-model TabbyML/Mistral-7B
```

Thanks

**Comment by wsxiaoys** [comment]
Glad you figured it out! Seems the root cause is cuda runtime version unmatch.
